File: March_2023/605.Can_Place_Flowers.py
# LeetCode 605. Can Place Flowers
# beginner friendly way

# in C++ availabel in --> 605.Can_Place_Flowers.cpp

# Date: March-20-2023

class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:

        for i in range(len(flowerbed)):
            if flowerbed[i] == 0 and (i == 0 or flowerbed[i-1] == 0) and (i == len(flowerbed) - 1 or flowerbed[i+1] == 0):
                flowerbed[i] = 1;
                n -= 1

                if n == 0:
                    return True
        
        if n > 0:
            return False

        return True
    

File: March_2023/605. Can_Place_Flowers.cpp
// LeetCode 605. Can Place Flowers
// beginner friendly way

// in python availabel in --> 605.Can_Place_Flowers.py

// Date: March-20-2023

#include <iostream>
#include<vector>
using namespace std;

class Solution
{

public:
    bool canPlaceFlowers(vector<int> flowerbed, int n)
    {

        if (n == 0)
        {
            return true;
        }
        // int count = 0;

        for (int i = 0; i < flowerbed.size(); i++)
        {

            if ((flowerbed[i] == 0) && (i == 0 || flowerbed[i - 1] == 0) && (i == flowerbed.size() - 1 || flowerbed[i + 1] == 0))
            {
                flowerbed[i] = 1;
                // count++;
                n--;

                if (n == 0)
                {
                    return true;
                }
            }

            // if(count >= n){
            //     return true;
            // }
        }

        if (n > 0)
        {
            return false;
        }

        return true;
    }
};

int main(){
    return 0;
}